### ISSUE TITLE
Add support for authentication using API Keys

### ISSUE THREAD

**Issue by igox** [MarlBurroW/Streamdeck-Uptime-Kuma#7]
Hi,

At the moment the plugin only supports login/password authentication.
My Kuma instance is set up with 2FA authentication and does not allow connection with login/password only.

It would be nice to add the possibility to set up plugin connection with API key.

Best,
iGoX

**Comment by keywal** [comment]
Also this please! :)

**Comment by CommanderStorm** [comment]
Logging in via api keys is currently not a feature offered. (see https://github.com/louislam/uptime-kuma/issues/3625)
The API keys currently only provide access to the `/metrics` endpoint and will be used for the api, once uptime-kuma has one.

See these issues (all are blockers for using API-Keys like this):
- user management and permission management: see tracking issue https://github.com/louislam/uptime-kuma/issues/128
- Api see tracking issue https://github.com/louislam/uptime-kuma/issues/118
- implementing the nessesary APIs

**Comment by MarlBurroW** [comment]
I'm not leveraging the HTTP API. Instead, I tap into the socket delivering data to the official frontend. This strategy allows for real-time data retrieval without the need for consistent polling. However, it's not a method that's officially documented/supported, so I believe it's a kind ofworkaround or hack. I'll investigate to see if there are other authentication options beyond just username/password with this socket method.
Thanks for feedbacks

**Comment by MarlBurroW** [comment]
Alright, I've built a release (1.2.0) that supports two-factor authentication. I haven't published it on the official store yet because I need your feedback first. So, I'm relying on you to download the release and let me know if it works for you.

The update is significant as it also changes the authentication method. Previously, the login and password were stored in the plugin's settings. Now, those credentials are only used to retrieve the authentication token, which is stored instead. The only uncertainty is how long this token remains valid. If it has a time limit, authentication will no longer be possible once the token expires. I might then need to propose an update to the Uptime Kuma repository to allow for non-expiring tokens.

In short, I'll leave it to you to test. The flow is straightforward: when you test the connection and the server indicates it requires two-factor authentication, an additional field will appear for you to enter the temporary code from your authentication app. If the code is correct, you'll be able to save the settings.

**Comment by igox** [comment]
Hi @MarlBurroW,

I freshly installed the v1.2.0 release.
The authentication workflow worked flawlessly and Auth token has been retrieved without a glitch! 
The global settings have been saved and all my probes updated accordingly. 👍 

I will let run the plugin and see how it goes with the token validity period...
I'll do a feedback to let you know.

Thanks for this update!

iGoX

**Comment by MarlBurroW** [comment]
Thanks igox for your test !
Louislam has confirmed that tokens never expires, this feature is done.
I'll publish this update on the store.
